### DISCUSSION TITLE
x-show gets overridden by :style when the style is recalculated

### DISCUSSION THREAD

**Discussion by SRS69** [alpinejs/alpine#3360] [1. Help] [answered]
This problem should be related to [this other problem](https://github.com/alpinejs/alpine/issues/434) that has been solved.
The problem occurs when on an element with `x-show` and `:style` directives the style is recalculated. Since `x-show` is only a `display: none` when the `:style` is recalculated the `display: none` is completely overwritten.
I did an example on [CodeSandbox](https://codesandbox.io/embed/alpinejs-x-show-and-style-bug-u2hi4s). Initially, when you load the page the text is not visible but when you press the button, which only changes the color of the text with `x-bind:style`, so by recalculating the style, even if the expression in the `x-show` is false the element is still shown.

**Comment by ekwoka** [comment, marked as answer]
You need to use the object binding syntax to make them work together. alpine does not parse your string for key value pairs so it it just applies the style as provides when it's a string

https://alpinejs.dev/directives/bind#binding-styles
